### ISSUE TITLE
Add Flower Baseline: FedAvg+CIFAR-10

### ISSUE THREAD

**Issue by charlesbvll** [adap/flower#1541]
### Paper

McMahan et al., 2016, Communication-Efficient Learning of Deep Networks from Decentralized Data, CIFAR-10

### Link

https://arxiv.org/abs/1602.05629

### Maybe give motivations about why the paper should be implemented as a baseline.

This paper was the first to propose a federated approach for deep learning on decentralized data.

### For first time contributors

- [ ]  Read the [`first contribution` doc](https://flower.dev/docs/first-time-contributors.html)
- [ ]  Complete the Flower tutorial
- [ ]  Read the Flower Baselines docs to get an overview:
    - [ ]  [https://flower.dev/docs/using-baselines.html](https://flower.dev/docs/using-baselines.html)
    - [ ]  [https://flower.dev/docs/contributing-baselines.html](https://flower.dev/docs/contributing-baselines.html)

### Prepare - understand the scope

- [ ] Read the paper linked above
- [ ] Create the directory structure in Flower Baselines (just the `__init__.py` files and a `README.md`)
- [ ] Before starting to write code, write down all of the specs of this experiment in a README (dataset, partitioning, model, number of clients, all hyperparameters, …)
- [ ] Open a draft PR

### Implement - make it work

- [ ] Implement some form of dataset loading and partitioning in a separate `dataset.py` (doesn’t have to match the paper exactly)
- [ ] Implement the model in PyTorch
- [ ] Write a test that shows that the model has the number of parameters mentioned in the paper
- [ ] Implement the federated learning setup outlined in the paper, maybe starting with fewer clients
- [ ] Plot accuracy and loss
- [ ] Run it and check if the model starts to converge

### Align - make it converge

- [ ] Implement the exact data partitioning outlined in the paper
- [ ] Use the exact hyperparameters outlined in the paper
- [ ] Make it converge to roughly the same accuracy that the paper states
- [ ] Commit the final hyperparameters and plots
- [ ] Mark the PR as ready

**Comment by Emmaka9** [comment]
Is this issue open or closed? If it is open, I would like to work on it.

**Comment by charlesbvll** [comment]
> Is this issue open or closed? If it is open, I would like to work on it.

Hi @Emmaka9 ! Unfortunately this issue is outdated, as the format of baselines as changed since then. I will close this as not planned

**Comment by Emmaka9** [comment]
I am looking to familiarize myself better with flwr by contributing. Let me know if there is an open good first issue. @charlesbvll

**Comment by charlesbvll** [comment]
@Emmaka9 You should check out this [page](https://flower.ai/docs/framework/contributor-ref-good-first-contributions.html) for a few pointers, but things that are not captured there and that are great as a first contribution are new strategies. I will make sure to update the docs to reflect this.
